net/usrsock/usrsock_conn.c

```diff
@@ -45,7 +46,9 @@
 
 /* The array containing all usrsock connections. */
 
+#ifndef CONFIG_NET_ALLOC_CONNS
 static struct usrsock_conn_s g_usrsock_connections[CONFIG_NET_USRSOCK_CONNS];
+#endif
```

Comment:
What about
```
int usrsock_connidx(FAR struct usrsock_conn_s *conn)
{
  int idx = conn - g_usrsock_connections;

  DEBUGASSERT(idx >= 0);
  DEBUGASSERT(idx < ARRAY_SIZE(g_usrsock_connections));

  return idx;
}
```
???

Reply:
this function will be removed from #5154